### PR TITLE
Update language codes

### PR DESCRIPTION
**Pull request by scttnlsn**
This switches to using language codes with regional dialects (i.e. `en_US`, `zh_TW`, etc.)

### CODE REVIEW

**Comment by connieimdialog** [comment]
@scttnlsn 
spa should be: es  (that one exists in the list, yes?)
aar is a mistake. can you just set that to ... es as well?

**Comment by scttnlsn** [comment]
@connieimdialog Great, thanks.  Yes `es` exists.

File: app/admin/checklists.rb [hunk 1]
```diff
@@ -42,7 +42,7 @@
   end
 
   show do
-    Globalize.with_locale :eng do
+    Globalize.with_locale :en_US do
```
Comment (sapraaman) [inline app/admin/checklists.rb]: why the change @scttnlsn ?

Comment (scttnlsn) [reply app/admin/checklists.rb]: Just to keep things consistent.  All the YAML locales are of this form and we would otherwise need to keep a mapping between the two schemes.

**Comment by scttnlsn** [comment]
@sapraaman Locales are using this new form since that's what Transifex outputs.  The value stored in a user's language preference field are now the new form: i.e. `en_US`  I removed all the 3-letter codes (which we were using previously) and we no longer need to map from 3-letter to 2-letter since the new form is used everywhere.

File: app/models/user.rb [hunk 2]
```diff
@@ -21,7 +21,7 @@
 #  invitations_count      :integer          default(0)
 #  approved               :boolean          default(FALSE), not null
 #  author_id              :string
-#  language               :string           default("eng"), not null
+#  language               :string           default("en_US"), not null
```
Comment (sapraaman) [inline app/models/user.rb]: @scttnlsn can this cause any issue in the existing data ?

Comment (scttnlsn) [reply app/models/user.rb]: @sapraaman Connie had told me it was OK if everyone's language is reset.  Since `eng` will be undefined in the new scheme, the default of `en_US` will be used.  We could alternatively run a migration to explicitly reset values to `en_US`

**Comment by connieimdialog** [comment]
hi! noticing that fr.yml comes up only when "French-France" is selected but not when "French" is.  can i create a duplicate of these files, and can one be tied to French and one to French-France, @scttnlsn

**Comment by scttnlsn** [comment]
@connieimdialog Yes indeed!  We could also change the language mapping file (https://github.com/globalvoices/NewsFrames-Code/blob/develop/config/languages.csv) so that both point to the same YAML file (i.e. change https://github.com/globalvoices/NewsFrames-Code/blob/develop/config/languages.csv#L165 to `French, fr_FR`)

**Comment by connieimdialog** [comment]
@scttnlsn i have created fr_FR.yml and devise.fr_FR.yml and tweaked the fr:/fr_FR: accordingly. if this works, then can close this and push everything to PROD

**Comment by connieimdialog** [comment]
yes! looks good
